src/command/info/info.c

```diff
@@ -81,30 +85,121 @@ STRING_STATIC(INFO_STANZA_STATUS_MESSAGE_MISSING_STANZA_PATH_STR,   "missing sta
+STRING_STATIC(INFO_STANZA_STATUS_MESSAGE_PG_MISMATCH_STR,           "database mismatch across repos");
 
-STRING_STATIC(INFO_STANZA_STATUS_MESSAGE_LOCK_BACKUP_STR,           "backup/expire running");
+#define INFO_STANZA_STATUS_MESSAGE_LOCK_BACKUP                      "backup/expire running"
 
 /***********************************************************************************************************************************
-Set error status code and message for the stanza to the code and message passed
+Data types and structures
+***********************************************************************************************************************************/
+// Repository information for a stanza
+typedef struct InfoRepoData
+{
+    unsigned int key;                                               // User-defined repo key
+    CipherType cipher;                                              // Encryption type (0 = none)
+    const String *cipherPass;                                       // Passphrase if the repo is encrypted (else NULL)
+    int stanzaStatus;                                               // Status code of the the stanza on this repo
+    unsigned int backupIdx;                                         // Index of the next backup that may be a candidate for sorting
+    InfoBackup *backupInfo;                                         // Contents of the backup.info file of the stanza on this repo
+    InfoArchive *archiveInfo;                                       // Contents of the archive.info file of the stanza on this repo
+} InfoRepoData;
+
+#define FUNCTION_LOG_INFO_REPO_DATA_TYPE                                                                                           \
+    InfoRepoData *
+#define FUNCTION_LOG_INFO_REPO_DATA_FORMAT(value, buffer, bufferSize)                                                              \
+    objToLog(value, "InfoRepoData", buffer, bufferSize)
+
+// Stanza with repository list of information for each repository
+typedef struct InfoStanzaRepo
+{
+    const String *name;                                             // Name of the stanza
+    uint64_t currentPgSystemId;                                     // Current postgres system id for the stanza
+    unsigned int currentPgVersion;                                  // Current postgres version for the stanza
+    InfoRepoData *repoList;                                         // List of configured repositories
+} InfoStanzaRepo;
+
+#define FUNCTION_LOG_INFO_STANZA_REPO_TYPE                                                                                         \
+    InfoStanzaRepo *
+#define FUNCTION_LOG_INFO_STANZA_REPO_FORMAT(value, buffer, bufferSize)                                                            \
+    objToLog(value, "InfoStanzaRepo", buffer, bufferSize)
+
+// Group all databases with the same system-id and version together regardless of db-id or repo
+typedef struct DbGroup
+{
+    uint64_t systemId;
```

Comment:
Comments for these struct members?

Reply:
fixed 8af20b6e1cacc2b4b68cbe5a4b7ab891ffa48768